components/service/glean/src/main/java/mozilla/components/service/glean/CommonMetricData.kt

```diff
@@ -4,15 +4,34 @@
 
 package mozilla.components.service.glean
 
+import mozilla.components.support.base.log.logger.Logger
+
 /**
  * This defines the common set of data shared across all the different
  * metric types.
  */
 interface CommonMetricData {
-    val applicationProperty: Boolean?
-    val disabled: Boolean?
-    val group: String?
-    val name: String?
-    val sendInPings: List<String>?
-    val userProperty: Boolean?
+    val applicationProperty: Boolean
+    val disabled: Boolean
+    val category: String
+    val name: String
+    val sendInPings: List<String>
+    val userProperty: Boolean
+}
+
+public fun shouldRecord(metric: CommonMetricData, logger: Logger): Boolean {
```

Comment:
This could be a protected function on the `CommonMetricData`. What do you think?

Reply:
Sure -- I didn't realise interfaces could have method implementations, but it seems they can.  I agree that would be cleaner.